### PR TITLE
SWTCH-972 no error while returning to login screen

### PR DESCRIPTION
**Pull request by DawidGil-jlabs**
Steps to reproduce problem:
Go to `login` method in IamService. Before or after initializing connection with waller provider add line `await Promise.reject()` which will imitate the error.

Solution for this is to stop reloading page, just display error message and clear user data from Store and IamService. Whenever user tries to login he will be able to see this error if it will occur.

I also cleaned up a little bit components and IamService from linter errors.

### CODE REVIEW

File: src/app/shared/services/iam.service.ts [hunk 1]
```diff
@@ -116,58 +121,48 @@ export class IamService {
...
-      });
-    }
-    else {
-      this._user.next(undefined);
-    }
+    this.store.dispatch(userClaimsActions.setDidDocument({didDocument: this._didDocument}));
+    this.store.dispatch(userClaimsActions.loadUserClaims());
+  }
+
+  private get isUserSettedUp(): boolean {
```
Comment (jrhender) [inline src/app/shared/services/iam.service.ts:150]: Slightly better grammar
```suggestion
  private get isUserSetUp(): boolean {
```

File: src/app/shared/services/iam.service.ts [hunk 2]
```diff
@@ -116,58 +121,48 @@ export class IamService {
 
           retVal = true;
         }
-      }
-      catch (e) {
+      } catch (e) {
         console.error(e);
-        this.logout();
-        location.reload();
+        this.toastr.error('Could not login, please try again later');
```
Comment (jrhender) [inline src/app/shared/services/iam.service.ts:126]: @jcmelati What do you think of this message?

**Review by jrhender — APPROVED** [review]
Looks good to me. So we're not sure what the root cause of the login errors are but this will stop the reloading upon error, right?

File: src/app/shared/services/iam.service.ts [hunk 3]
```diff
@@ -192,11 +188,8 @@ export class IamService {
     return this._iam;
   }
 
-  get userProfile() {
-    return this._user.asObservable();
-  }
-
-  setUserProfile(data: any) {
+  // TODO: remove in future this method and _user property in favor of the store.
```
[the remark this answers is not in the document]
Comment (DawidGil-jlabs) [reply src/app/shared/services/iam.service.ts:191]: I have two ideas. 
First longer one would take some time and i would need to move few things to store + a lot of testing to be sure that i didn't broke anything. - this is approach that i would like to go with in future. 
Second one was fast but didn't work out or i am doing something wrong 🤔 In getter `isUserPresent` i wanted to do sth like this `return Boolean(await this.store.select(getUserProfile).toPromise())` but maybe i'm messing something with promises. I will try again with it.

[the remark this answers is not in the document]
Comment (DawidGil-jlabs) [reply src/app/shared/services/iam.service.ts]: Ok, i found a solution to this. I needed to add an operator `take(1)` to make it work. So from now on user profile will be handled in store.

[the remark this answers is not in the document]
Comment (DawidGil-jlabs) [reply src/app/shared/services/iam.service.ts]: And I did the same with `_didDocument` property in IamService :) It's not ideal for me but it's better than it was. @jrhender thanks for motivation!

**Pull request merged**